src/cache.js

```diff
@@ -15,33 +15,30 @@ const isUpdated = (a, b, properties) => {
 }
 
 const validateDate = (cache) => {
-  if ((cache.timeoutInSec !== null && cache.timeOfDay !== null) ||
-     (cache.timeOfDay !== null && cache.expiryDate !== null) ||
-     (cache.timeoutInSec !== null && cache.expiryDate !== null)) {
-      console.log('You can only have 1 expirySettings option. Set to default 300 seconds')
+  if (((cache.timeoutInSec !== null || undefined) && (cache.timeOfDay !== null || undefined)) ||
+     ((cache.timeOfDay !== null || undefined) && (cache.expiryDate !== null || undefined)) ||
+     ((cache.timeoutInSec !== null || undefined) && (cache.expiryDate !== null || undefined))) {
+    console.log('You can only have 1 expirySettings option. Set to default 300 seconds')
     return {'timeoutInSec': {'value': 300}, 'valuesNull': cache.expirySettings.valuesNull}
   } else if (cache.expiryDate) {
-      return {'expiryDate': {'value': cache.expiryDate}, 'valuesNull': cache.expirySettings.valuesNull}
```

Comment:
Door de check hierboven kun je denk ik gewoon het volledige object retourneren. Alles wat null / undefined is zal apigee toch negeren.
`
return  {
'expiryDate': { 'value': cache.expiryDate }, 
'timeOfDay': {'value': cache.timeOfDay},
'timeoutInSec': {'value': cache.timeoutInSec},
'valuesNull': cache.expirySettings.valuesNull
}
`